onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.cc

```diff
@@ -107,6 +111,276 @@ std::shared_ptr<KernelRegistry> GetMklDnnKernelRegistry() {
 }
 }  // namespace mkl_dnn
 
+bool MKLDNNExecutionProvider::RunSubgraph(const onnxruntime::GraphViewer& graph_viewer, 
+                                          const std::vector<const KernelRegistry*>& kernel_registries,
+                                          std::vector<std::unique_ptr<ComputeCapability>>& result) const {
+  // switch between mkldnn-vanilla and mkldnn-subgraph implementation using
+  // MKLDNN_SUBGRAPH environment variable
+  bool use_subgraph = true;
+
+  const char* env = getenv("ORT_MKLDNN_SUBGRAPH");
+  int use_subgraph_env = 0;
+  if (env != nullptr)
+    use_subgraph_env = atoi(env);
+
+  if (use_subgraph_env == 0) {
+    use_subgraph = false;
+    for (auto& node : graph_viewer.Nodes()) {
+      for (auto registry : kernel_registries) {
+        if (registry->TryFindKernel(node, Type()) != nullptr) {
+          std::unique_ptr<IndexedSubGraph> sub_graph = std::make_unique<IndexedSubGraph>();
+          sub_graph->nodes.push_back(node.Index());
+          result.push_back(std::make_unique<ComputeCapability>(std::move(sub_graph)));
+          break;
+        }
+      }
+    }
+  }
+  return use_subgraph;
+}
+
+std::vector<std::unique_ptr<ComputeCapability>> MKLDNNExecutionProvider::GetCapability(
+    const onnxruntime::GraphViewer& graph_viewer,
+    const std::vector<const KernelRegistry*>& kernel_registries) const {
+  ORT_UNUSED_PARAMETER(kernel_registries);
+  std::vector<std::unique_ptr<ComputeCapability>> result;
+
+  // temporary switch to toggle between mkldnn-vanilla and mkldnn-subgraph implementation using
+  // ORT_MKLDNN_SUBGRAPH environment variable
+  if (RunSubgraph(graph_viewer, kernel_registries, result) == false){
+    return result;
+  }
+
+  // use sub-graph implementation
+  std::vector<std::string> mkl_ops = {"Conv", "BatchNormalization", "Relu", "Sum",
```

Comment:
gets lost if kept as a local variable here. should be part of the class definition? 
and it's probably better to use a set instead of vector.